pkg/apimachinery/registered/registered.go

```diff
@@ -114,22 +114,24 @@ func IsEnabledVersion(v unversioned.GroupVersion) bool {
 	return found
 }
 
-// EnabledVersions returns all enabled versions.
-func EnabledVersions() (ret []unversioned.GroupVersion) {
-	for v := range enabledVersions {
-		ret = append(ret, v)
+// EnabledVersions returns all enabled versions.  Groups are randomly ordered, but versions within groups
+// are priority order from best to worst
+func EnabledVersions() []unversioned.GroupVersion {
+	ret := []unversioned.GroupVersion{}
+	for _, groupMeta := range groupMetaMap {
+		ret = append(ret, groupMeta.GroupVersions...)
```

Comment:
> I agree on the version ordering.  I'm just trying to understand why enabled
> versions is random.  It's fine for this issue, but I don't know that I
> consider randomness correct in this context.

I'm not sure that I know what order I'd like this to have.  I know that I don't like alphabetical, because everyone will name theirs `aaaa`.  Given package loading ordering which isn't obvious to me, I'm not sure that I like enablement order.

I think I want group ordering to be suggested by discover endpoints and strictly ordered client-side if desired.